crates/oci-distribution/src/client.rs

```diff
@@ -145,8 +226,13 @@ impl Client {
             None => return Ok(()),
         };
 
-        // Right now, we do read-only auth.
-        let pull_perms = format!("repository:{}:pull", image.repository());
+        // Allow for either push or pull authentication
+        let scope = match operation {
+            RegistryOperation::Pull => format!("repository:{}:pull", image.repository()),
+            RegistryOperation::Push => format!("repository:{}:push", image.repository()),
+            RegistryOperation::PushPull => format!("repository:{}:pull,push", image.repository()),
```

Comment:
I based my decision here from https://docs.docker.com/registry/spec/auth/token/#requesting-a-token, which I read as you can supply your scopes, and then the authentication takes the intersect of the scopes you provide and what you're allowed to use. `PushPull` was useful for my roundtrip test, I'd imagine _most_ scenarios you would probably use one or the other

Reply:
@bacongobbler After doing some implementation locally with `wash`, it appears that the `Push` operation alone is insufficient to push an image to a registry. https://github.com/deislabs/krustlet/pull/453/files/c536fe3b547b85821ba68d6cd88efab769d8d961#diff-1ed88c51082d2507f4329c76729010e254880b8fffd0661f06fc042aa7325e1bR151 must be `PushPull`, or a solo push will not work. I didn't notice this in my test as I was doing the authentication beforehand rather than leaving it up to the `push` function. I'll need to do some more research here to figure out why, but I'm going to fix it in the code for now